cibuildwheel/util.py

```diff
@@ -58,11 +60,25 @@ def read_python_configs(config: PlatformName) -> List[Dict[str, str]]:
 
 
 class BuildSelector:
-    def __init__(self, build_config: str, skip_config: str):
+    def __init__(self, build_config: str, skip_config: str, *, requires_python: Optional[SpecifierSet] = None):
         self.build_patterns = build_config.split()
         self.skip_patterns = skip_config.split()
+        self.requires_python = requires_python
 
     def __call__(self, build_id: str) -> bool:
+        # Filter build selectors by python_requires if set
+        if self.requires_python is not None and '-' in build_id:
+            py_ver_str = build_id.split('-')[0]
+            try:
+                major = int(py_ver_str[2])
+                minor = int(py_ver_str[3:])
+            except ValueError:
+                pass
+            else:
+                version = Version(f"{major}.{minor}")
+                if not self.requires_python.contains(version):
+                    return False
```

Comment:
> Maybe a package specifies python_requires `>=3.9.1` because they rely on a bug fix?

That would never happen! What [crappy project maintainers](https://github.com/pybind/pybind11/pull/2576) would test the Python release candidates so badly that they [allow this to happen](https://github.com/pybind/pybind11/commit/2a263e087c0aa7a70a6ffc6dcfbfc0d3a5b9a2fa)? _cough cough_

Reply:
`3.9.0 == 3.9` is true, sadly `3.9.1 == 3.9` is false (I thought it was true at one point so I likely have a few bad `version_requires` lying around; you really do need to write `!=3.4.*`).

> test the Python release candidates so badly 

Never! I'm sure it would show up [months before release](https://github.com/pybind/pybind11/issues/2391#issue-678783700)...